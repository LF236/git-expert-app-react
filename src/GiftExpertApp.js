import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';
const GiftExpertApp = ({ defaultCategories = [] }) => {
    const [categories, setCategories] = useState( defaultCategories );

    // const handleAdd = () => {
    //     setCategories( [ ...categories, 'Dead Note' ] );
    //     // setCategories(cats => [ ...cats, 'Deadth Note' ]);
    // };

    return (
        <main>
            <h2>GiftExpertApp</h2>
                <AddCategory setCategories={ setCategories } />
            <hr />
            
            <div>
                {
                    categories.map(category => {
                        return <GifGrid category={ category } key={ category }/>
                    })
                }

            </div>

        </main>
    )
}

export default GiftExpertApp;