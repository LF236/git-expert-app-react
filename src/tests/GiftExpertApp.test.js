import { shallow } from 'enzyme';
import GiftExpertApp from '../GiftExpertApp';
describe( 'Pruebas sobre el componente <GiftExpertApp />', () => {
    test( 'El componente renderizado debe ser igual a su Snapshot por defecto', () => {
        const wrapper = shallow( <GiftExpertApp /> );
        expect( wrapper ).toMatchSnapshot();
    } );

    test( 'El componente renderizado debe tener mostrar una lista de categorias', () => {
        let listaCategorias = [ 'Naruto', 'Death Note' ];
        const wrapper = shallow( <GiftExpertApp defaultCategories={ listaCategorias }/> );
        expect( wrapper.find( 'GifGrid' ).length ).toBe( listaCategorias.length );
    } );
} )