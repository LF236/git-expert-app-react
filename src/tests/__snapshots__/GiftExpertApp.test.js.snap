// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GiftExpertApp /> El componente renderizado debe ser igual a su Snapshot por defecto 1`] = `
<main>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div />
</main>
`;
